src/FESpaces/ConformingFESpaces.jl

```diff
@@ -114,15 +116,15 @@ end
 Generate a CellFE from a vector of reference fes
 """
 function CellFE(
-  cell_map::AbstractArray{<:Field},
-  cell_reffe::AbstractArray{<:ReferenceFE})
-
+  model::DiscreteModel,
+  cell_reffe::AbstractArray{<:ReferenceFE},
+ )
   ctype_reffe, cell_ctype = compress_cell_data(cell_reffe)
   ctype_num_dofs = map(num_dofs,ctype_reffe)
   ctype_ldof_comp = map(reffe->get_dof_to_comp(reffe),ctype_reffe)
   cell_conformity = CellConformity(cell_reffe)
-  cell_shapefuns = lazy_map(get_shapefuns,cell_reffe,cell_map)
-  cell_dof_basis = lazy_map(get_dof_basis,cell_reffe,cell_map)
+  cell_shapefuns = get_shapefuns(model,cell_reffe)
```

Comment:
Do you need to pass a full discrete model here? is not enough to do something like 

```
cell_shapefuns = lazy_map(get_shapefuns,cell_reffe,cell_map, cell_flip)
```
?

Reply:
> Do you need to pass a full discrete model here?

We shall agree that there is some work to do at the global FE space level to compute the (cell-wise restriction) of global shape functions and global DoFs in the case of global H(div)-conforming FE spaces (that, e.g., is not required to be done for global grad-conforming FE spaces). In the case of your code snippet, it would be to build the `cell_flip` array out of the model. In the case of the code I have written, those lines of code which I added to `ConformingFESpaces.jl`. The code which is in charge of doing this work cannot go to the `ReferenceFEs` module, otherwise we would have a cycle. I guess that you are trying to avoid this cycle by having interfaces that do not depend on model, and that thus can go to `ReferenceFEs.jl`. But I am not sure if this is a good solution, because of reasons which I try to make clear below. 
 
I understand the concern of having code in `ConformingFESpaces.jl` which depends on the particular kind of global FE space at hand. In order to solve this, I propose to add a new source file, `DivConformingFEspaces.jl`, in FESpaces. See 
https://github.com/gridap/Gridap.jl/pull/579/files#diff-ea5ed427dd3dd1fad920c1f2014e8950d5866705bde29f7278c6d93390ae72c1R1 for a motivation. In a nutshell, this source file implements those customizations of `get_shape_funs` and `get_dof_basis` for the global conforming FE space at hand. 

>  is not enough to do something like 
> ```
> cell_shapefuns = lazy_map(get_shapefuns,cell_reffe,cell_map, cell_flip)
> ```
> ?

I am not sure if this would be convenient. 

First, you would be passing the `cell_flip` array in order to build the (cell-wise restriction) of global shape functions for all global conforming FE spaces, even if it is not needed (e.g., for Grad-Confirming spaces).

Second, it is hard to anticipate if `cell_flip` is the info which will be needed in order to compute the (cell-wise restriction) of global shape functions for those global conforming FE spaces which we may want to implement  in the future. 

By having a `get_shapefuns` interface with model, and the ReferenceFE, we have more flexibility in defining the global shape functions according to the global FE space at hand. If the name is confusing, we can perhaps think of another name, instead of reusing the one that is used in order to compute the reference FE shape functions. 

Let me know what do you think.